### PR TITLE
improve benchmark script

### PR DESCRIPTION
**Pull request by MartinNowak**
- simpler old runbench script
- best of 10 for more stable timings
- align and cleanup output

### CODE REVIEW

**Comment by MartinNowak** [comment]
> It helps to add scoping curly braces around the usage of lines so the file is closed before being renamed.

done

**Comment by MartinNowak** [comment]
> Passing --DRT arguments to the benchmark is a bit ugly, though. Using quotes and an additional space as in " --DRT-gcopt=profile:1" works, but we might want to pass them differently and convert them (e.g. ---DRT-gcopt) or use rt_args to forward them from runbench to the test.

What's the problem here?
`rdmd runbench -v gcbench -m64 -release -- --DRT-gcopt=profile:1`

**Comment by rainers** [comment]
If you use a current runtime to compile runbench (and not 2.066 as probably your rdmd does), it will remove `--DRT-gcopt=profile:1` before main (https://github.com/D-Programming-Language/druntime/blob/master/src/rt/dmain2.d#L372) and cannot pass it to the tests.

**Comment by rainers** [comment]
> What's the problem here?

Actually, with rdmd compiled from git-head,  `--DRT-gcopt=profile:1` will not even be passed to runbench, but will print GC statistics for rdmd.

**Comment by MartinNowak** [comment]
> Actually, with rdmd compiled from git-head, --DRT-gcopt=profile:1 will not even be passed to runbench, but will print GC statistics for rdmd.

Ah right, didn't thought of that, solved now.

**Comment by rainers** [comment]
> Ah right, didn't thought of that, solved now.

Looks good for runbench, but you won't be able to use rdmd in the future to build and run it.

File: benchmark/runbench.d [hunk 1]
```diff
@@ -335,254 +45,154 @@ int main(string[] args)
...
     {
-        writefln("!!! [DISABLED: %s]", testArgs.disabled_reason);
-        return 0;
-    }
+        import std.datetime, std.algorithm : min;
+        auto sw = StopWatch(AutoStart.yes);
+        auto minDur = Duration.max;
 
-    removeIfExists(output_file);
+        stdout.writef("RUNNING %-20s", bin.relativePath(bindir));
```
Comment (rainers) [inline benchmark/runbench.d]: Can we please reduce the vertical space here for better alignment and more possible output by the GC profiling, e.g. 
```
stdout.writef("R %-16s", bin.baseName.stripExtension);
```

Comment (MartinNowak) [inline benchmark/runbench.d]: done

**Comment by MartinNowak** [comment]
> Looks good for runbench, but you won't be able to use rdmd in the future to build and run it.

That's OK for me `dmd -run` works just as well.

**Comment by rainers** [comment]
> That's OK for me dmd -run works just as well.

Ok, then let's get it in.

**Comment by rainers** [comment]
Auto-merge toggled on

**Comment by rainers** [comment]
Just noticed: quotes around runtime arguments are lost, which causes trouble if they contains spaces. This can be fixed in a later PR, though.

**Comment by braddr** [comment]
The auto-tester automatically turns off auto-merge approval when a pull request is updated with a new commit.  This is on purpose to make sure changes are reviewed.  I thought it posted when it did that though, and those are missing from this request.  I'll have to go look through the logs.

**Comment by rainers** [comment]
Auto-merge toggled on

**Comment by schveiguy** [comment]
> The auto-tester automatically turns off auto-merge approval when a pull request is updated with a new commit.

A case where this should have happened and did not (I wasn't aware of this feature):

https://github.com/D-Programming-Language/dmd/pull/4239

Auto merge only turned on once, early on. I made 2 more commits after that, and never got turned off.

**Comment by schveiguy** [comment]
And a case where it was mysteriously turned off without a commit: https://github.com/D-Programming-Language/druntime/pull/1082

**Comment by braddr** [comment]
Thanks for those references.  I've looked at the logs and filed 2 bugs for me to address.

**Comment by MartinNowak** [comment]
> Thanks for those references. I've looked at the logs and filed 2 bugs for me to address.

Thanks @braddr, can you actually help me out little to test the auto-tester scripts locally?
I want to add dlang.org support too, because the documentation breaks way too often.
https://github.com/braddr/d-tester/issues/41#issuecomment-65947240

**Pull request merged**